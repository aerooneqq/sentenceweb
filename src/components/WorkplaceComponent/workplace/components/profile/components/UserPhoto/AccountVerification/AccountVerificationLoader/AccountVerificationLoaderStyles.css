.accountVerificationLoaderCont { 
    width: 100%;
    height: 30px; 

    display: flex; 
    flex-direction: row;

    justify-content: center; 
    align-items: center;

    box-sizing: border-box;
}

.accountVerificationLoaderIcon { 
    width: 25px;
    height: 25px;

    background: green;

    border-radius: 25px;
}

.accountVerificationLoaderText { 
    width: 70%;
    height: 25px;

    margin-left: 10px;

    background: #e0e0e0;

    border-radius: 5px;
}