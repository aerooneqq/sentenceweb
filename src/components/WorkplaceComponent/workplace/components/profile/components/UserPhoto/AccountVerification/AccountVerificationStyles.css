#accountVerificationOutterContainer{ 
    width: 100%;
    height: auto;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    margin-top: 30px;
}

#accountVerificationCodeInput{ 
    width: 120px;
    height: 30px;

    box-sizing: border-box;

    padding-left: 3px;
    padding-right: 3px;

    border-width: 1px;
    border-style: solid;
    border-color: orangered;
    border-radius: 5px;

    font-family: Arial;
    font-size: 15px;
    color: gray;
}

    #accountVerificationCodeInput:focus{ 
        outline: none;
    }

#accountVerificationText{
    box-sizing: border-box;
    
    font-family: Arial; 
    color: black;
    font-size: 14px;

    padding-left: 10%;
    padding-right: 10%;

    margin-bottom: 10px;
}

#enterCodeContainer{ 
    display: flex; 
    flex-direction: row; 
    justify-content: center; 
    align-items: center;
}

#sendConfirmationCodeBtn{ 
    width: 70px;
    height: 30px;

    border-style: solid; 
    border-width: 1px;
    border-color: orange; 
    border-radius: 5px;

    background: white; 

    margin-left: 10px;

    font-size: 15px;
    color: orange;
}

    #sendConfirmationCodeBtn:focus{ 
        outline: none;
    }

    #sendConfirmationCodeBtn:hover{ 
        color: orangered;
        border-color: orangered;
    }

#verifiedAccountContainer { 
    width: 100%;
    height: auto; 

    display: flex;
    flex-direction: row;

    justify-content: center;
    align-items: center;
}

#verifiedAccountIcon { 
    width: 30px; 
    height: 30px;
}

#verifiedAccountText { 
    margin-left: 10px;

    font-size: 14px; 
    color: black;
}