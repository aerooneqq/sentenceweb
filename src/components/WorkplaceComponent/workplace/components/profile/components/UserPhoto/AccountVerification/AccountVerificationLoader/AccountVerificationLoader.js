import React, {Component} from "react";

//Styles
import "./AccountVerificationLoaderStyles.css"

export default class AccountVerificationLoader extends Component { 

    constructor(props) { 
        super(props);
    }

    render() { 
        return ( 
            <div className = "accountVerificationLoaderCont">

                <div className = "accountVerificationLoaderIcon" />
                <div className = "accountVerificationLoaderText" />

            </div>
        )
    }

}