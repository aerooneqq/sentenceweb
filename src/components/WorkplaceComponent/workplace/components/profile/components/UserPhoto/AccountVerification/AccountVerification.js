import React, {Component} from "react";

//Styles
import "./AccountVerificationStyles.css";

//Services
import CodesService from "../../../../../../../../services/CodesServices/CodesService";
import UserService from "../../../../../../../../services/UserServices/UserService";

//Icons
import verifiedAccountIcon from "./img/user_account_verified.png";

//Components
import Loader from "../../../../../../../loader/Loader";

//App messages
import {alertAppMessage} from "../../../../../../../ApplicationMessage/ApplicationMessageManager";


export default class AccountVerification extends Component { 
    constructor(props){ 
        super(props);

        this.codesService = new CodesService(localStorage.getItem("token"));
        this.userService = new UserService(localStorage.getItem("token"));

        this.state = { 
            inputValue: "",
            isLoading: true,
            isAccVerified: null
        };

        this.handleInputValueChange = this.handleInputValueChange.bind(this);
        this.handleSendCodeClick = this.handleSendCodeClick.bind(this);
        this.checkIfAccVerified = this.checkIfAccVerified.bind(this);
    }

    componentDidMount() { 
        this.checkIfAccVerified(); 
    } 

    checkIfAccVerified() { 
        this.userService.getPartialData(["isAccountVerified"])
            .then(res => { 
                this.setState({ 
                    isLoading: false,
                    isAccVerified: res.data.isAccountVerified
                });
            }).catch(er => { 
                if (er.reponse) { 
                    alertAppMessage(er.response.data, "error");
                }
            });
    }

    handleInputValueChange(event) {
        this.setState({ 
            inputValue: event.target.value
        });
    }

    handleSendCodeClick(event) { 
        let code = document.getElementById("accountVerificationCodeInput").value;

        this.setState({ 
            isLoading: true
        });

        this.codesService.activateAccount(code).then(res => { 
            if (res.status === 200) { 
                alertAppMessage("The account is verified.", "success");
                this.checkIfAccVerified();
            } 
        }).catch(er => { 
            if (er.response) { 
                alertAppMessage(er.response.data, "error");
            }
            else { 
                alertAppMessage("Unknown error did happen.", "error");
            }

            this.setState({
                isLoading: false
            });
        });
    }

    render() { 
        return (
            <div id = "accountVerificationOutterContainer">
                {this.state.isLoading === true ? 
                    <Loader message = "Getting verification data..." />
                    : 
                    this.state.isAccVerified === true ? 
                        <div id = "verifiedAccountContainer">
                            <img src = {verifiedAccountIcon} id = "verifiedAccountIcon"
                                alt = "Account verified"/>
                            <div id = "verifiedAccountText">
                                Your account is verified
                            </div>
                        </div>
                    :
                        <div>
                            <div id = "accountVerificationText">
                                Enter the code from the last e-mail. If you don't verify your account, it will be blocked.
                            </div>
                            <div id = "enterCodeContainer">
                                <input id = "accountVerificationCodeInput" type = "text"
                                    onChange = {this.handleInputValueChange} />
                                <button id = "sendConfirmationCodeBtn" onClick = {this.handleSendCodeClick}>
                                    Send
                                </button>
                            </div>
                        </div>
                }

            </div>
        );
    }
}