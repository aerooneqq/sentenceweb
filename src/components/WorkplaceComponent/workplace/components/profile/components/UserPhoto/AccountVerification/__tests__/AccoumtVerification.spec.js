import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import AccountVerification from "../AccountVerification";
import AccountVerificationLoader from "../AccountVerificationLoader/AccountVerificationLoader";

describe("Account verification", () => { 
    it("should render correctly after the state has changed", () => { 
        const accountVerification = shallow(<AccountVerification />);
        accountVerification.setState({ 
            isLoading: false,
            isFirstLoad: false,
        });

        expect(shallowToJson(accountVerification)).toMatchSnapshot();
    });

    it("should have AccountVerificationLoader when first loaded", () => { 
        const accountVerification = shallow(<AccountVerification />);
        expect(accountVerification.find("div#accountVerificationOutterContainer")
            .find(AccountVerificationLoader).exists()).toBe(true);
    });

    it("should have proper initial state", () => { 
        const accountVerification = shallow(<AccountVerification />);
        const properState = { 
            isLoading: true,
            isFirstLoad: true,
            isAccVerified: null,
            inputValue: ""
        }

        for (let key in properState) { 
            expect(properState[key]).toBe(accountVerification.state(key));
        }
    });

    it("chould call checkIfAccVerified when componentDidMount", () => { 
        const spy = jest.spyOn(AccountVerification.prototype, "checkIfAccVerified").mockImplementation(() => {});
        const accountVerification = shallow(<AccountVerification />);

        expect(spy).toBeCalled();
    })
});