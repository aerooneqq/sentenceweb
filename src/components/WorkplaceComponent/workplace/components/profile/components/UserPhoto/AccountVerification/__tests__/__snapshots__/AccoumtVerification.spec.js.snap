// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account verification should render correctly after the state has changed 1`] = `
<div
  id="accountVerificationOutterContainer"
>
  <div>
    <div
      id="accountVerificationText"
    >
      Enter the code from the last e-mail. If you don't verify your account, it will be blocked.
    </div>
    <div
      id="enterCodeContainer"
    >
      <input
        id="accountVerificationCodeInput"
        onChange={[Function]}
        type="text"
      />
      <button
        id="sendConfirmationCodeBtn"
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </div>
</div>
`;
